pagination/tests/test_querystrings.py

```diff
@@ -0,0 +1,38 @@
+from django.http import QueryDict
+
+from .utils import RequestTestCase
+from .. import querystrings
+
+
+class TestQuerystringUpdate(RequestTestCase):
+    def setUp(self):
+        """Create a request and put some GET args in it."""
+        self.request = self.create_request()
+        self.request.GET = QueryDict('page=3&q=search_term')
+
+    def test_unchanged_querystring(self):
+        """Test that the method returns an unchanged request.GET if kwargs is empty."""
+        querystring = querystrings.querystring_update(self.request)
+
+        # Re-parse the result into a dict because the query arguments don't appear in
+        # a predictable order.
+        new_querydict = QueryDict(querystring).dict()
+        expected = {
+            'page': '3',
+            'q': 'search_term',
+        }
+        self.assertDictEqual(new_querydict, expected)
+
+    def test_querystring_update(self):
+        """Test that individual GET arguments are updated or preserved correctly."""
+        search_term = 'a_different_search_term'
+        querystring = querystrings.querystring_update(self.request, q=search_term)
+
+        # Re-parse the result into a dict because the query arguments don't appear in
+        # a predictable order.
+        new_querydict = QueryDict(querystring).dict()
+        expected = {
+            'page': '3',
+            'q': search_term,
+        }
+        self.assertDictEqual(new_querydict, expected)
```

Comment:
`assertSequenceEqual` is also better than `assertEqual` for list-esque things, since it doesn't mind if one item is a list and the other is a tuple (a common occurrence).  (https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertSequenceEqual)  I seem to remember it also doing better at comparing items within the sequence, but I might be wrong on that one.